underscore.js

```diff
@@ -371,7 +363,19 @@
       if (!isArrayLike(obj)) obj = _.values(obj);
       return obj[_.random(obj.length - 1)];
     }
-    return _.shuffle(obj).slice(0, Math.max(0, n));
+
+    var set = isArrayLike(obj) ? obj : _.values(obj);
+    var length = getLength(set);
+    n = Math.max(Math.min(n, getLength(set)), 0);
+    var sample = _.clone(set);
+    var rand, temp;
+    for (var index = length - 1; index > length - n - 1; index--) {
```

Comment:
sure, but why not use a for loop for that? Iterating between two integers is the textbook example for when to use ```for``` instead of ```while```.